test/JDBC/expected/parallel_query/with_recompile-vu-verify.out

```diff
@@ -0,0 +1,3450 @@
+              Recheck Cond: (c41 = 1)
+              ->  Bitmap Index Scan on tb_recomp_41_pkey
+                    Index Cond: (c41 = 1)
+  Query Text: select * from tb_recomp_41 where c41 = 1
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Index Scan using tb_recomp_41_pkey on tb_recomp_41
+              Index Cond: (c41 = 1)
+Query Text: EXEC    p_recomp_61 1
+  Query Text: select * from tb_recomp_61 where c61 = "@p"
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Bitmap Heap Scan on tb_recomp_61
+              Recheck Cond: (c61 = 1)
+              ->  Bitmap Index Scan on tb_recomp_61_pkey
+                    Index Cond: (c61 = 1)
+  Query Text: select * from tb_recomp_61 where c61 = 1
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Index Scan using tb_recomp_61_pkey on tb_recomp_61
+              Index Cond: (c61 = 1)
+~~END~~
+
+set babelfish_showplan_all off
+go
+
+select '--- EXEC with RECOMPILE: same as EXEC without RECOMPILE ----'
+go
+~~START~~
+varchar
+--- EXEC with RECOMPILE: same as EXEC without RECOMPILE ----
+~~END~~
+
+set babelfish_showplan_all on
+go
+exec    p_recomp_41 1 with recompile
+exec    p_recomp_61 1 with recompile
+go
+~~START~~
+text
+Query Text: EXEC    p_recomp_41 1 /* EXECUTE WITH RECOMPILE */
+  Query Text: select * from tb_recomp_41 where c41 = "@p"
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Bitmap Heap Scan on tb_recomp_41
+              Recheck Cond: (c41 = 1)
+              ->  Bitmap Index Scan on tb_recomp_41_pkey
+                    Index Cond: (c41 = 1)
+  Query Text: select * from tb_recomp_41 where c41 = 1
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Index Scan using tb_recomp_41_pkey on tb_recomp_41
+              Index Cond: (c41 = 1)
+Query Text: EXEC    p_recomp_61 1 /* EXECUTE WITH RECOMPILE */
+  Query Text: select * from tb_recomp_61 where c61 = "@p"
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Bitmap Heap Scan on tb_recomp_61
+              Recheck Cond: (c61 = 1)
+              ->  Bitmap Index Scan on tb_recomp_61_pkey
+                    Index Cond: (c61 = 1)
+  Query Text: select * from tb_recomp_61 where c61 = 1
+  ->  Gather
+        Workers Planned: 1
+        Single Copy: true
+        ->  Index Scan using tb_recomp_61_pkey on tb_recomp_61
+              Index Cond: (c61 = 1)
+~~END~~
+
+set babelfish_showplan_all off
+go
+
+select '--- turning on index scan ----'
+go
+~~START~~
+varchar
+--- turning on index scan ----
+~~END~~
+
+select set_config('enable_indexscan', 'on', false)
+go
+~~START~~
+text
+on
+~~END~~
+
+select current_setting('enable_indexscan') as indexscan_on
+go
+~~START~~
+text
+on
+~~END~~
+
+
+select '--- EXEC without RECOMPILE: still creating recompiled plan on every execution: index scan for both statements ----'
```

Comment:
I'd like to learn from your experience whether there were any needs for `exec without RECOMPILE` to re-enable plan caching even though the procedure was created by `with RECOMPILE` option.

Reply:
T-SQL does not provide such a mechanism. When a procedure is created with RECOMPILE, the only way to stop it from doing a recompile on every exec is to recreate the proc without RECOMPILE or run ALTER PROCEDURE without RECOMPILE. 
Deciding to use RECOMPILE is done after performance tuning and query plan debugging and reflects a conscious decision about overriding the regular query plan behaviour.